Format generated config files using the short array syntax

cykirsch opened the pull request:
<!--- Provide a general summary of the Pull Request in the Title above -->

### Description
<!--- Provide a description of the changes proposed in the pull request -->
1. Magento is following a standard of new code using short array syntax (https://github.com/magento/magento2/issues/758).
2. Magento has an automated checker to enforce the short array syntax precommit (https://github.com/magento/magento2/blob/2.2/.php_cs.dist#L31). 
3. Magento recommends checking in `app/etc/config.php` (http://devdocs.magento.com/guides/v2.2/config-guide/config/config-php.html)
>Check this file in to source control and use it in your development, staging, and production systems.
4. Magento's auto-generated `app/etc/config.php` uses the long array syntax.

Put that all together and the auto-generated config doesn't match the required standards to be checked in. The change made in this PR is to modify `PhpFormatter` to recursively return the array representation using short array syntax `[]` instead of long `array()`. If the given variable is not an array, it uses the standard `var_export` behavior.

### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
1. bin/magento setup:upgrade
2. app/etc/config.php should use the short array syntax

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

cykirsch committed:
Format generated config files using the short array syntax

cykirsch commented:
As an added note, I did attempt to do this using `Zend\Code\Generator\ValueGenerator` instead of `var_export`, but the array depth wasn't being tracked properly. The unit test array would look like this

    [
      'ns1' => [
            's1' => [
                's11',
                's12',
            ],
            's2' => [
                's21',
                's22',
            ],
        ],
      'ns2' => [
            's1' => [
                's11',
            ],
        ],
      'ns3' => 'just text',
      'ns4' => 'just text',
    ]

cykirsch committed:
Fix the config formatter scenarios with $comments to also use short array syntax

cykirsch committed:
Fix code style issues

ishakhsuvarov commented on lib/internal/Magento/Framework/App/DeploymentConfig/Writer/PhpFormatter.php:
```diff
@@ -51,18 +53,42 @@ private function formatData($data, $comments = [], $prefix = '  ')
+    {
+        if (gettype($var) === 'array') {
+            $indexed = array_keys($var) === range(0, count($var) - 1);
+            $r = [];
```
Please avoid variables with unclear names. `$r` does not really describe what is it for.

ishakhsuvarov commented on lib/internal/Magento/Framework/App/DeploymentConfig/Writer/PhpFormatter.php:
```diff
@@ -51,18 +53,42 @@ private function formatData($data, $comments = [], $prefix = '  ')
+     * @param integer $depth
+     * @return string
+     */
+    private function varExportShort($var, $depth = 0)
```
Please take advantage of PHP 7.0 syntax (type hints and return types for methods)

ishakhsuvarov commented on lib/internal/Magento/Framework/App/DeploymentConfig/Writer/PhpFormatter.php:
```diff
@@ -51,18 +53,42 @@ private function formatData($data, $comments = [], $prefix = '  ')
+     */
+    private function varExportShort($var, $depth = 0)
+    {
+        if (gettype($var) === 'array') {
```
Is there any specific reason not to use `is_array`?
Additionally, readability may be slightly improved by reversing the logic like

```php
if (!is_array($var)) {
    return var_export($var, true);
}
/// Transforming to short array syntax here
...
```

ishakhsuvarov commented on lib/internal/Magento/Framework/App/DeploymentConfig/Writer/PhpFormatter.php:
```diff
@@ -11,6 +11,8 @@
  */
 class PhpFormatter implements FormatterInterface
 {
+    const INDENT = '  ';
```
Please add a description for the constant (docBlock)

cykirsch committed:
Fixes per ishakhsuvarov code review:
- const documenation
- reverse condition for readability
- better variable name
- type and return hints

cykirsch commented:
@ishakhsuvarov @okorshenko I was more or less expecting these changes would be in 2.2.2. Is it odd to you that it is on the 2.2.2-preview branch but not on the 2.2.2 tag? Or does that happen often?

clementblanco commented:
What if we want 4 spaces tab indentation?

cykirsch commented:
@clementblanco That did cross my mind, and I would prefer it as well, but I neglected to do it...I think because I was continuing to use `var_export` and I didn't want to mix 4 and 2.
